src/download.rs

```diff
@@ -18,47 +22,130 @@ struct LatestReleaseApiResponse {
+                        version
+                    } else {
+                        "v".to_string() + &version
+                    }
+                }
+                None => {
+                    let latest_tag_url = match name {
+                        "forc" => format!(
+                            "{}{}/{}",
+                            GITHUB_API_REPOS_BASE_URL, SWAY_REPO, RELEASES_LATEST
+                        ),
+                        "fuel-core" => format!(
+                            "{}{}/{}",
+                            GITHUB_API_REPOS_BASE_URL, FUEL_CORE_REPO, RELEASES_LATEST
+                        ),
+                        "fuelup" => format!(
+                            "{}{}/{}",
+                            GITHUB_API_REPOS_BASE_URL, FUELUP_REPO, RELEASES_LATEST
+                        ),
+                        _ => bail!("Unrecognized component: {}", name),
+                    };
+                    if let Ok(result) = get_latest_tag(&latest_tag_url) {
+                        result
+                    } else {
+                        bail!("Error getting latest tag for component: {}", name);
+                    }
+                }
+            },
+            release_url: match name {
+                "forc" => SWAY_RELEASE_DOWNLOAD_URL.to_string(),
+                "fuel-core" => FUEL_CORE_RELEASE_DOWNLOAD_URL.to_string(),
+                "fuelup" => FUELUP_RELEASE_DOWNLOAD_URL.to_string(),
+                _ => bail!("Unrecognized component: {}", name),
+            },
+        })
+    }
+}
 
-    let os = match std::env::consts::OS {
-        "macos" => "darwin",
-        "linux" => "linux-gnu",
-        unsupported_os => bail!("Unsupported os: {}", unsupported_os),
-    };
+pub fn tarball_name(download_cfg: &DownloadCfg) -> Result<String> {
+    match download_cfg.name.as_ref() {
+        "forc" => {
+            let os = match std::env::consts::OS {
+                "macos" => "darwin",
+                "linux" => "linux",
+                unsupported_os => bail!("Unsupported os: {}", unsupported_os),
+            };
+            let architecture = match std::env::consts::ARCH {
+                "aarch64" => "arm64",
+                "x86_64" => "amd64",
+                unsupported_arch => bail!("Unsupported architecture: {}", unsupported_arch),
+            };
+
+            Ok(format!("forc-binaries-{}_{}.tar.gz", os, architecture))
+        }
 
-    Ok(format!(
-        "fuel-core-{}-{}-{}-{}.tar.gz",
-        // strip the 'v' from the version string to match the file name of the releases
-        &version[1..version.len()],
-        architecture,
-        vendor,
-        os
-    ))
+        "fuel-core" => {
+            let architecture = match std::env::consts::ARCH {
+                "aarch64" => "aarch64",
+                "x86_64" => "x86_64",
+                unsupported_arch => bail!("Unsupported architecture: {}", unsupported_arch),
+            };
```

Comment:
Could also be:

```rust
let architecture = match std::env::consts::ARCH {
    "aarch64" | "x86_64" => std::env::consts::ARCH,
    unsupported_arch => bail!("Unsupported architecture: {}", unsupported_arch),
};
```

Reply:
wait, I could've sworn you made the same comment on a previous PR and I made this change before 😂

Reply:
I remember now, it was in the `fuelup self update` PR I think